pkg/utils/k8sutil.go

```diff
@@ -983,9 +985,7 @@ func ensureFuncJob(client kubernetes.Interface, funcObj *spec.Function, or []met
 			return err
 		}
 		_, err = client.BatchV2alpha1().CronJobs(funcObj.Metadata.Namespace).Patch(job.Name, types.StrategicMergePatchType, data)
-		if err != nil {
-			return err
-		}
+		return err
```

Comment:
so what happens when the error is nil ?

Reply:
that is the issue, `err` is redefined inside this `if` scope, so if `err` is `nil` here it wasn't returned and the final value is the one outside the `if` (that is not `nil`)